api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -228,27 +229,35 @@ private boolean foundBlobIdsContainsUserRecentlyModifiedResource(
   }
 
   @VisibleForTesting
-  public boolean hasValidBlobIdIfNotebookNamePresent(
-      DbUserRecentlyModifiedResource dbUserRecentResource) {
-    if (dbUserRecentResource.getResourceType()
-        == DbUserRecentlyModifiedResource.DbUserRecentlyModifiedResourceType.NOTEBOOK) {
-      return Optional.ofNullable(dbUserRecentResource.getResourceId())
-          .map(name -> uriToBlobId(name).isPresent())
-          .orElse(true);
+  public boolean isValidResource(DbUserRecentlyModifiedResource resource) {
+    // handle null and unparseable resourceIds
```

Comment:
Notebooks have URL Strings, but other resources should have parseable-as-long Strings, like "123"